src/DETHRACE/common/flicplay.c

```diff
@@ -1365,30 +1397,33 @@ int LoadFlic(int pIndex) {
     PathCat(the_path, the_path, gMain_flic_list[pIndex].file_name);
     f = DRfopen(the_path, "rb");
 
-    if (!f) {
+    if (f == NULL) {
         FatalError(13, gMain_flic_list[pIndex].file_name);
     }
 
     gMain_flic_list[pIndex].the_size = GetFileLength(f);
     gMain_flic_list[pIndex].data_ptr = BrMemAllocate(gMain_flic_list[pIndex].the_size, 0x90u);
 
-    if (gMain_flic_list[pIndex].data_ptr) {
-        fread(gMain_flic_list[pIndex].data_ptr, 1u, gMain_flic_list[pIndex].the_size, f);
-        strcpy(gLast_flic_name, gMain_flic_list[pIndex].file_name);
-        fclose(f);
-        return 1;
-    } else {
+    if (gMain_flic_list[pIndex].data_ptr == NULL) {
         if (AllocationErrorsAreFatal()) {
             FatalError(14, gMain_flic_list[pIndex].file_name);
         }
+#ifdef DETHRACE_FIX_BUGS
+        fclose(f);
```

Comment:
is this to make the compiler happy?

Reply:
Not really.
If allocation errors are not fatal, this code path can leak.
Very unlikely, I know.